src/main/fc/fc_init.c

```diff
@@ -247,6 +247,16 @@ void init(void)
 #endif
 
     printfSupportInit();
+    
+    initEEPROM();
```

Comment:
Agreed. The better approach might be to start up the device, store a 0xDEADBEEF semi permanent type flag, and soft reboot to get the clock speed desired (so it is detected and done in the init function prior to main).

The draw back is the double boot on power up - but may be suitable.